--- configure.ac	2019-06-28 15:01:40.000000000 +0200
+++ configure.ac	2019-06-28 14:57:52.000000000 +0200
@@ -860,7 +860,7 @@
 	AC_SEARCH_LIBS([ERR_peek_error], [crypto], ,
 	               AC_MSG_ERROR([libcrypto not found]))
 
-	AC_SEARCH_LIBS([SSL_library_init], [ssl], ,
+	AC_SEARCH_LIBS([OPENSSL_init_ssl], [ssl], ,
 		[
 			if test x"$enable_shared" = x"yes"
 			then
@@ -874,9 +874,9 @@
 
 			# avoid caching issue - last result of SSL_library_init
 			# shouldn't be cached for this next check
-			unset ac_cv_search_SSL_library_init
+			unset ac_cv_search_OPENSSL_init_ssl_init
 			LIBCRYPTO_LIBS="$LIBCRYPTO_LIBS -ldl"
-			AC_SEARCH_LIBS([SSL_library_init], [ssl], ,
+			AC_SEARCH_LIBS([OPENSSL_init_ssl], [ssl], ,
 			               AC_MSG_ERROR([libssl not found]), [-ldl])
 		]
 	)
